server/notifications/Publisher.ts

```diff
@@ -0,0 +1,26 @@
+interface ISubscription {
+	[key: string]: () => void;
+}
+
+export default class Publisher {
+	subscriptions: ISubscription;
+	constructor(subscriptions: ISubscription) {
+		this.subscriptions = subscriptions;
+	}
+
+	subscribe(postId: string, fn: () => void) {
+		this.subscriptions[postId] = fn;
+	}
```

Comment:
In case you want to keep this class as universal as you can (storing only the functions to invoke), so I'd recommend avoiding specifics. Like, change `postId` to `id`.